src/main/java/org/alliancegenome/curation_api/services/GeneOntologyAnnotationService.java

```diff
@@ -106,14 +109,24 @@ private Long getGOID(GeneOntologyAnnotationDTO uiEntity) {
 		return goID;
 	}
 
-	private Species getSpecies(String orgAbbreviation) {
-		if (species != null) {
+	private List<Species> getSpecies(String orgAbbreviation) {
+		if (CollectionUtils.isNotEmpty(species)) {
 			return species;
 		}
+		if (orgAbbreviation.equalsIgnoreCase("XB")) {
+			species = new ArrayList<>();
+			species.add(getSingleSpecies("XBXL"));
```

Comment:
The whole getSingleSpecies method could be replaced here with `speciesDAO.findByField("displayName", "XB").getSingleResult();`